android/src/main/java/com/applovin/reactnative/AppLovinMAXAdView.java

```diff
@@ -155,66 +164,86 @@ public void attachAdView()
                 {
-                    adView = new MaxAdView( adUnitId, adFormat, AppLovinMAXModule.getInstance().getSdk(), currentActivity );
-                    adView.setListener( AppLovinMAXModule.getInstance() );
-                    adView.setRevenueListener( AppLovinMAXModule.getInstance() );
+                    adView.setPlacement( placement );
+                }
 
-                    if ( placement != null )
-                    {
-                        adView.setPlacement( placement );
-                    }
+                if ( customData != null )
+                {
+                    adView.setCustomData( customData );
+                }
 
-                    if ( customData != null )
-                    {
-                        adView.setCustomData( customData );
-                    }
+                if ( adaptiveBannerEnabled != null )
+                {
+                    adView.setExtraParameter( "adaptive_banner", adaptiveBannerEnabled ? "true" : "false" );
+                }
 
-                    if ( adaptiveBannerEnabled != null )
+                // Set this extra parameter to work around a SDK bug that ignores calls to stopAutoRefresh()
+                adView.setExtraParameter( "allow_pause_auto_refresh_immediately", "true" );
+
+                if ( autoRefresh != null )
+                {
+                    if ( autoRefresh )
                     {
-                        adView.setExtraParameter( "adaptive_banner", adaptiveBannerEnabled ? "true" : "false" );
+                        adView.startAutoRefresh();
                     }
-
-                    // Set this extra parameter to work around a SDK bug that ignores calls to stopAutoRefresh()
-                    adView.setExtraParameter( "allow_pause_auto_refresh_immediately", "true" );
-
-                    if ( autoRefresh != null )
+                    else
                     {
-                        if ( autoRefresh.booleanValue() )
-                        {
-                            adView.startAutoRefresh();
-                        }
-                        else
-                        {
-                            adView.stopAutoRefresh();
-                        }
+                        adView.stopAutoRefresh();
                     }
+                }
 
-                    adView.loadAd();
+                adView.loadAd();
 
-                    currentWidthPx = getWidth();
-                    currentHeightPx = getHeight();
+                currentWidthPx = getWidth();
+                currentHeightPx = getHeight();
 
-                    addView( adView );
-                }
+                addView( adView );
             }
-        } );
+        }, 250 );
     }
 
     void destroy()
     {
         if ( adView != null )
         {
+            AppLovinMAXModule.d( "Unmounting MAAdView: " + adView );
+
             removeView( adView );
 
             adView.setListener( null );
```

Comment:
We should also set `adView` to `null` in Java, and `nil` on iOS since we're destroying it on un-mount

Reply:
Updated both.